### PR TITLE
[Merged by Bors] - feat: independence of functions under conditioning

### PR DESCRIPTION
**Pull request by YaelDillies**
From PFR

---

- [x] depends on: #18663

Do people think this should go in a new file?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Probability/ConditionalProbability.lean
Comment (kex-y) [inline Mathlib/Probability/ConditionalProbability.lean:252]: Can you use the `μ[| X ← x]` notation on the RHS

File: Mathlib/Probability/ConditionalProbability.lean
Comment (kex-y) [inline Mathlib/Probability/ConditionalProbability.lean:268]:
```suggestion
    μ[⋂ i ∈ s, f i |⋂ i, Y i ⁻¹' t i] = ∏ i ∈ s, μ[f i|Y i in t i] := by
```
Does this work?

Comment (YaelDillies) [reply Mathlib/Probability/ConditionalProbability.lean:268]: Whoops sorry I had already done this change locally, but somehow reverted it before pushing

File: Mathlib/Probability/ConditionalProbability.lean
Comment (kex-y) [inline Mathlib/Probability/ConditionalProbability.lean:312]: If `μ (Y i ⁻¹' t i) = 0` for some `i` then the conditional measure is the zero measure and the statement should still hold true right?

Comment (kex-y) [reply Mathlib/Probability/ConditionalProbability.lean:312]: Same comment for the previous lemma

Comment (YaelDillies) [reply Mathlib/Probability/ConditionalProbability.lean:312]: The zero measure is not independent (at least not according to mathlib's definition) because `0 (⋂ i ∈ ∅, f i) ≠ ∏ i ∈ ∅, 0 (f i)`

**Comment by kex-y** [comment]
Thanks!

maintainer merge

**Comment by kim-em** [comment]
This PR significantly increases imports to many files. I would strongly prefer that material is moved to new files instead.

And if there's a good reason that can't / shouldn't happen, then it needs to be discussed, at least in the PR description, ideally even in module doc-strings, before we should merge.

**Comment by YaelDillies** [comment]
I was hoping a probability theorist (@RemyDegenne ? @kex-y ?) would come and tell where they would like those lemmas to live.

**Comment by kex-y** [comment]
I suppose you can put these lemmas in Probability/Independence/Conditional

**Comment by YaelDillies** [comment]
Turns out moving things to `Probability.Independence.Basic` works just fine. I've taken the opportunity to add the kernel version of the first lemma (the second lemma I don't think one can state yet), but I can't deduce the one measure version from it because the sigma-algebra fed into `iIndepFun` is of a specific shape.

File: Mathlib/Probability/Independence/Kernel.lean
Comment (RemyDegenne) [inline Mathlib/Probability/Independence/Kernel.lean:1265]: That comment refers to a completely undefined lemma (unless you happen to know that a similarly name lemma exists in another file). Please make the reference more precise or remove it.

**Review by RemyDegenne — APPROVED** [review]
LGTM
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**
